src/tagmanager/tm_workspace.c

```diff
@@ -828,20 +828,22 @@ static GHashTable *get_includes(TMSourceFile *source, GPtrArray **header_candida
 		gchar *hdr_name = g_path_get_basename(hdr_tag->name);
 		GPtrArray *tm_files = g_hash_table_lookup(theWorkspace->source_file_map, hdr_name);
 
-		if (tm_files)
+		if (tm_files && tm_files->len > 0)
 		{
 			guint j;
 
-			for (j = 0; j < tm_files->len; j++)
+			if (!*header_candidates)
```

Comment:
There isn't any caller that passes non-NULL, right? I would define `get_includes()` such that it always writes to `*header_candidates` (unless there are none).

Reply:
The check
```
if (!*header_candidates)
```
is to avoid doing the comparisons below if we already set `header_candidates` in the previous iteration. Also notice `*header_candidates = NULL;` at the beginning of the function.

What you describe would be
```
if (!header_candidates)
```
(i.e. without the dereference)

Reply:
You are right, I didn't notice the outer loop. So I guess the idea is to avoid multiple assignments to `*header_candidates` in case a file is included multiple times (maybe legitimately, via different directories)?

So I think this is alright now.

Reply:
> So I guess the idea is to avoid multiple assignments to *header_candidates in case a file is included multiple times (maybe legitimately, via different directories)?

This is just an optimization so the code inside the `if` block doesn't have to be executed if we already found `header_candidates`.